[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46355 [quant][eagermode] Add additional_fuser_method_mapping to config
* **#46346 [quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern**

Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

jerryzh168 committed:
[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern

Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

vkuzo commented on torch/quantization/quantize_fx.py:
```diff
@@ -184,6 +184,17 @@ def prepare_fx(model, qconfig_dict, inplace=False, prepare_custom_config_dict=No
         },
+
+        # Additional fusion patterns
+        "additioanl_fusion_pattern": {
```
nit: typo?

vkuzo reviewed (approved):
lg, should we add a test?

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 commented:
> lg, should we add a test?

was thinking of adding them when we have the first real use case so that we don't need to make things up.

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern"


Summary:
Allow user to provide additional fusion/quant patterns for fx graph mode

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317437](https://our.internmc.facebook.com/intern/diff/D24317437)

[ghstack-poisoned]
